### ISSUE TITLE
bcrypt version @5.1.0 errors out with bun install

### ISSUE THREAD

**Issue by deval77** [oven-sh/bun#4063]
### What platform is your computer?

Linux 5.15.0-1037-aws x86_64 x86_64

### How did you attempt to install or upgrade?

installed all the deps. with command - bun install 

after installation, when the app started with both commands ( node and bun ). 

### What do you see instead?

**WITH NODE** 
node:internal/modules/cjs/loader:936
  throw err;
  ^

Error: Cannot find module '/home/rbian_raj_1/devalProjects/backend_code/node_modules/bcrypt/lib/binding/napi-v3/bcrypt_lib.node'
Require stack:
- /home/rbian_raj_1/devalProjects/backend_code/node_modules/bcrypt/bcrypt.js
- /home/rbian_raj_1/devalProjects/backend_code/utils/commonUtils.js
- /home/rbian_raj_1/devalProjects/backend_code/service/chat/index.js
- /home/rbian_raj_1/devalProjects/backend_code/controller/chat/index.js
- /home/rbian_raj_1/devalProjects/backend_code/routes/chat/index.js
- /home/rbian_raj_1/devalProjects/backend_code/routes/index.js
- /home/rbian_raj_1/devalProjects/backend_code/my_project.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/home/rbian_raj_1/devalProjects/backend_code/node_modules/bcrypt/bcrypt.js:6:16)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1155:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/home/rbian_raj_1/devalProjects/backend_code/utils/commonUtils.js:2:16)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1155:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/home/rbian_raj_1/devalProjects/backend_code/service/chat/index.js:19:40)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1155:10)
    at Module.load (node:internal/modules/cjs/loader:981:32) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/home/rbian_raj_1/devalProjects/backend_code/node_modules/bcrypt/bcrypt.js',
    '/home/rbian_raj_1/devalProjects/backend_code/utils/commonUtils.js',
    '/home/rbian_raj_1/devalProjects/backend_code/service/chat/index.js',
    '/home/rbian_raj_1/devalProjects/backend_code/controller/chat/index.js',
    '/home/rbian_raj_1/devalProjects/backend_code/routes/chat/index.js',
    '/home/rbian_raj_1/devalProjects/backend_code/routes/index.js',
    '/home/rbian_raj_1/devalProjects/backend_code/my_project.js'
  ]
}
[nodemon] app crashed - waiting for file changes before starting...

**WITH BUN**
Cannot find module "/home/rbian_raj_1/devalProjects/backend_code/node_modules/bcrypt/lib/binding/napi-v3/bcrypt_lib.node" from "/home/rbian_raj_1/devalProjects/backend_code/node_modules/bcrypt/bcrypt.js"

**Comment by iambpn** [comment]
+1 same issue

**Comment by geekypandey** [comment]
+1 on the same.
I tried installing packages with `npm` and then running `bun run dev`, it was working then.

Same issue at: https://github.com/oven-sh/bun/issues/299

**Comment by nithincvpoyyil** [comment]
I'm also getting the same issue, I guess this is happening due to [bcrypt's native library bindings](https://forum.serverless.com/t/after-making-a-deploy-i-test-the-function-and-show-this/8624/2). 
However, I'm unsure about  how its working when running `bun dev` following a `npm install`

**Comment by iambpn** [comment]
@nithincvpoyyil i think the crypto module is not implemented in the bun yet so its causing this error.

**Comment by manastunga787** [comment]
it works with `npm install bcrypt`
but fails with `bun add bcrypt`

**Comment by iambpn** [comment]
I think since crypto is an internal module when installed through bun it resolves in different ways compared to nodejs. So it works through npm install but not with bun install.

**Comment by nmrusso** [comment]
Same issue here. I want to migrate from node to bun and this bug doesn't allow me to do so. I understand that bcrypt is a medium-specific library.

**Comment by kafkainthevoid** [comment]
this issue happened for years and they don't fix it, i think they want us to use bun hashing api
[https://bun.sh/docs/api/hashing](url)

**Comment by Jarred-Sumner** [comment]
There are two separate bugs here:
1) The install parts will be fixed by #4263. bun install isn't running the postinstall script to compile this dependency.
2) The runtime parts will need to be fixed differently. There is a napi issue causing a crash. We haven't narrowed it down yet.

In the meantime, I suggest using `Bun.password.verify` which is builtin bcrypt support in Bun.

**Comment by Electroid** [comment]
Duplicate of #4959

**Comment by Jarred-Sumner** [comment]
Fixed by @dylan-conway and @paperdave in #7132

This will be part of the Bun v1.0.17 release, which ships in a few hours from the time of writing.

**Comment by Ileies** [comment]
Even though it got fixed, on NixOS with Bun 1.1.37 I'm getting the exact same issue now. It works when using NodeJS.
